Backpack-SwiftUI/Calendar/Classes/Core/CalendarMonthGrid.swift

```diff
@@ -20,110 +20,78 @@ import SwiftUI
 
     @State private var dayCellHeight: CGFloat = 0
     @ViewBuilder let dayCell: (Date) -> DayCell
+    @ViewBuilder let disabledDayCell: (Date) -> DisabledDayCell
     @ViewBuilder let emptyLeadingDayCell: () -> EmptyLeadingDayCell
     @ViewBuilder let emptyTrailingDayCell: () -> EmptyTrailingDayCell
     @ViewBuilder let dayAccessoryView: (Date) -> DayAccessoryView
     
-    private let daysInAWeek = 7
+    let calculator: CalendarGridCalculator
     
     var body: some View {
-        let firstWeekday = calendar.firstWeekday // Locale-aware first day of the week
-        let weekdayOfMonthStart = calendar.component(.weekday, from: monthDate)
-        // Calculate the offset based on the first weekday
-        let daysFromPreviousMonth = (weekdayOfMonthStart - firstWeekday + daysInAWeek) % daysInAWeek
-
-        LazyVGrid(
-            columns: Array(repeating: GridItem(spacing: BPKSpacing.none.value), count: daysInAWeek),
-            spacing: BPKSpacing.lg.value
-        ) {
-            // Create cells for the days from the previous month that are shown in the first week of the current month.
-            previousEmptyCells(daysFromPreviousMonth: daysFromPreviousMonth)
-            let numberOfDaysInMonth = calendar.range(of: .day, in: .month, for: monthDate)!.count
-            // Create cells for the days in the current month
-            currentMonthDayCell(numberOfDaysInMonth: numberOfDaysInMonth)
-
-            // Create cells for the days from the next month that are shown in the last week of the current month
-            // The total number of cells used is the sum of the number of days in the current month and the number of
-            // days from the previous month that are shown
-            let totalCellsUsed = numberOfDaysInMonth + daysFromPreviousMonth
-            let remainingCells = daysInAWeek - (totalCellsUsed % daysInAWeek)
+        let grid = calculator.calculateCalendarGrid(monthDate: monthDate)
         
-            remainingEmptyCells(remainingCells: remainingCells)
+        VStack(alignment: .leading) {
+            ForEach(0..<grid.count, id: \.self) { row in
+                weekRow(grid: grid, row: row)
+            }
         }
     }
     
     @ViewBuilder
-    private func previousEmptyCells(daysFromPreviousMonth: Int) -> some View {
-        let preEmptyCells = Array(0..<daysFromPreviousMonth)
-            .map {
-                DayCellIdentifiable(id: "pre-\($0)\(monthDate)", index: $0)
-            }
-        ForEach(preEmptyCells) { _ in
-            VStack(spacing: BPKSpacing.none) {
-                emptyLeadingDayCell()
-                    .frame(height: dayCellHeight)
-                Spacer(minLength: BPKSpacing.none)
+    private func weekRow(grid: [[Date?]], row: Int) -> some View {
+        HStack(alignment: .top, spacing: BPKSpacing.none) {
+            ForEach(0..<grid[0].count, id: \.self) { col in
+                dayCellView(grid: grid, row: row, col: col)
```

Comment:
We simply go through the grid and render the corresponding cell, given the value